mlem/contrib/kubernetes/base.py

```diff
@@ -0,0 +1,281 @@
+import os
+from typing import ClassVar, List, Optional, Tuple
+
+from kubernetes import client, config
+
+from mlem.config import project_config
+from mlem.core.base import MlemABC
+from mlem.core.errors import DeploymentError, MlemError
+from mlem.core.objects import (
+    DeployState,
+    DeployStatus,
+    MlemBuilder,
+    MlemDeployment,
+    MlemEnv,
+    MlemModel,
+)
+from mlem.runtime.client import Client, HTTPClient
+from mlem.runtime.server import Server
+from mlem.ui import EMOJI_OK, echo
+
+from ..docker.base import (
+    DockerDaemon,
+    DockerImage,
+    DockerRegistry,
+    generate_docker_container_name,
+)
+from .build import build_k8s_docker
+from .context import K8sYamlBuildArgs, K8sYamlGenerator, ServiceTypeEnum
+from .utils import create_k8s_resources, namespace_deleted, pod_is_running
+
+POD_STATE_MAPPING = {
+    "Pending": DeployStatus.STARTING,
+    "Running": DeployStatus.RUNNING,
+    "Succeeded": DeployStatus.STOPPED,
+    "Failed": DeployStatus.CRASHED,
+    "Unknown": DeployStatus.UNKNOWN,
+}
+
+
+def find_index(nodes_list, node_name):
+    for i, each_node in enumerate(nodes_list):
+        if each_node.metadata.name == node_name:
+            return i
+    return -1
+
+
+class ServiceType(MlemABC):
+
+    abs_name: ClassVar = "k8s_service_type"
+    type: ClassVar = "kubernetes"
+
+    class Config:
+        type_root = True
+
+    def get_host_and_port(
+        self, service, namespace="mlem"  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[str], Optional[int]]:
+        """Returns host and port for the service in Kubernetes"""
+        return None, None
+
+
+class NodePortService(ServiceType):
+
+    type: ClassVar = "k8s_nodeport_service"
```

Comment:
you can just use "nodeport", "loadbalancer" etc since abs_type already has k8s service type keywords

Reply:
done